src/spikeinterface/extractors/sinapsrecordingh5extractor.py

```diff
@@ -0,0 +1,112 @@
+from pathlib import Path
+import numpy as np
+
+from ..core.core_tools import define_function_from_class
+from ..core import BaseRecording, BaseRecordingSegment
+
+try:
+    import h5py
+
+    HAVE_MCSH5 = True
+except ImportError:
+    HAVE_MCSH5 = False
+
+class SinapsResearchPlatformH5RecordingExtractor(BaseRecording):
+    extractor_name = "SinapsResearchPlatformH5"
+    mode = "file"
+    name = "sinaps_research_platform_h5"
+
+    def __init__(self, file_path):
+
+        assert self.installed, self.installation_mesg
+        self._file_path = file_path
+
+        mcs_info = openSiNAPSFile(self._file_path)
```

Comment:
Note that `mcs` stands for `multi-channel-system` ;)

Reply:
I know. I adapted the code from the MCS extractor :) It was the closest one.
SiNAPS is still a multi-channel-system, no?

Reply:
> I know. I adapted the code from the MCS extractor :) It was the closest one. SiNAPS is still a multi-channel-system, no?

Nope :) MCS is a german company that mainly makes in-vitro systems and other acquisition hardware.

Reply:
oh, ok. I understood this name just as it is. I will change the naming now, together with other edits.